packages/cursorless-vscode/src/ide/vscode/VscodeFileSystem.ts

```diff
@@ -1,52 +1,15 @@
-import {
-  Disposable,
-  FileSystem,
-  PathChangeListener,
-  walkFiles,
-} from "@cursorless/common";
-import { stat } from "fs/promises";
-import { max } from "lodash";
+import { Disposable, FileSystem, PathChangeListener } from "@cursorless/common";
+import { RelativePattern, Uri, workspace } from "vscode";
 
 export class VscodeFileSystem implements FileSystem {
   watchDir(path: string, onDidChange: PathChangeListener): Disposable {
-    // Just poll for now; we can take advantage of VSCode's sophisticated
-    // watcher later. Note that we would need to do a version check, as VSCode
-    // file watcher is only available in more recent versions of VSCode.
-    return new PollingFileSystemWatcher(path, onDidChange);
-  }
-}
-
-const CHECK_INTERVAL_MS = 1000;
-
-class PollingFileSystemWatcher implements Disposable {
-  private maxMtimeMs: number = -1;
-  private timer: NodeJS.Timer;
-
-  constructor(
-    private readonly path: string,
-    private readonly onDidChange: PathChangeListener,
-  ) {
-    this.checkForChanges = this.checkForChanges.bind(this);
-    this.timer = setInterval(this.checkForChanges, CHECK_INTERVAL_MS);
-  }
-
-  private async checkForChanges() {
-    const paths = await walkFiles(this.path);
-
-    const maxMtime =
-      max(
-        (await Promise.all(paths.map((file) => stat(file)))).map(
-          (stat) => stat.mtimeMs,
-        ),
-      ) ?? 0;
-
-    if (maxMtime > this.maxMtimeMs) {
-      this.maxMtimeMs = maxMtime;
-      this.onDidChange();
-    }
-  }
-
-  dispose() {
-    clearInterval(this.timer);
+    // FIXME: Support globs?
+    const watcher = workspace.createFileSystemWatcher(
+      new RelativePattern(Uri.file(path), "**"),
```

Comment:
You don't need to convert the string path to a uri. The first argument can just be a string.

Reply:
All of the examples [in the docs](https://code.visualstudio.com/api/references/vscode-api#RelativePattern:~:text=Out%20of%20workspace%20file%20watching) use `Uri.file`, so I'd be inclined to follow their lead

Reply:
huh interesting. I guess that's working so sure I'll change it here